sdks/go/test/integration/expansions_test.go

```diff
@@ -0,0 +1,169 @@
+)
+
+type testProcess struct {
+	killed bool
+	jar    string
+}
+
+func (p *testProcess) Kill() error {
+	p.killed = true
+	return nil
+}
+
+func failRun(_, _ string, _ ...string) (jars.Process, error) {
+	return nil, fmt.Errorf("unexpectedly running a jar, failing")
+}
+
+func succeedRun(_, jar string, _ ...string) (jars.Process, error) {
+	return &testProcess{jar: jar}, nil
+}
+
+// TestExpansionServices_GetAddr_Addresses tests calling GetAddr on provided addresses.
+func TestExpansionServices_GetAddr_Addresses(t *testing.T) {
+	addrsMap := map[string]string{
+		"label1": "testAddr1",
+		"label2": "testAddr2",
+		"label3": "testAddr3",
+	}
+	jarsMap := map[string]string{
+		"label2": "jarFilepath2",
+	}
+	es := &ExpansionServices{
+		addrs: addrsMap,
+		jars:  jarsMap,
+		procs: make([]jars.Process, 0),
+		run:   failRun,
+	}
+
+	// Ensure we get the same map we put in, and that addresses take priority over jars if
+	// both are given for the same label.
+	for label, wantAddr := range addrsMap {
+		gotAddr, err := es.GetAddr(label)
+		if err != nil {
+			t.Errorf("unexpected error when getting address for \"%v\": %v", label, err)
+			continue
+		}
+		if gotAddr != wantAddr {
+			t.Errorf("incorrect address for \"%v\", want %v, got %v", label, wantAddr, gotAddr)
+		}
+	}
+	// Check that nonexistent labels fail.
+	if _, err := es.GetAddr("nonexistent_label"); err == nil {
+		t.Errorf("did not receive error when calling GetAddr with nonexistent label")
+	}
+}
+
+// TestExpansionServices_GetAddr_Jars tests calling GetAddr on provided jars.
+func TestExpansionServices_GetAddr_Jars(t *testing.T) {
+	addrsMap := map[string]string{}
+	jarsMap := map[string]string{
+		"label1": "jarFilepath1",
+		"label2": "jarFilepath2",
+		"label3": "jarFilepath3",
+	}
+	es := &ExpansionServices{
+		addrs: addrsMap,
+		jars:  jarsMap,
+		procs: make([]jars.Process, 0),
+		run:   succeedRun,
+	}
+
+	// Call GetAddr on each jar twice, checking that the addresses remain consistent.
+	gotMap := make(map[string]string)
+	for label := range jarsMap {
+		gotAddr, err := es.GetAddr(label)
+		if err != nil {
+			t.Errorf("unexpected error when getting address for \"%v\": %v", label, err)
+			continue
+		}
+		gotMap[label] = gotAddr
+	}
+	for label, gotAddr := range gotMap {
+		secondAddr, err := es.GetAddr(label)
+		if err != nil {
+			t.Errorf("unexpected error when getting address for \"%v\": %v", label, err)
+			continue
+		}
+		if secondAddr != gotAddr {
+			t.Errorf("getAddr returned different address when called twice for \"%v\", "+
+				"attempt 1: %v, attempt 2: %v", label, gotAddr, secondAddr)
+		}
+	}
+	// Check that all jars were run.
+	gotJars := make([]string, 0)
+	for _, proc := range es.procs {
+		testProc := proc.(*testProcess)
+		gotJars = append(gotJars, testProc.jar)
+	}
+	wantJars := make([]string, 0)
+	for _, jar := range jarsMap {
+		wantJars = append(wantJars, jar)
+	}
+	if diff := cmp.Diff(wantJars, gotJars); diff != "" {
```

Comment:
You'll need to add the unordered compare here, *or* sort the slices first. TBH I'd just sort the slices.

Reply:
Done.